src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -103,6 +103,9 @@ public void save(AddressBook addressBook) throws StorageOperationException {
             final Marshaller marshaller = jaxbContext.createMarshaller();
```

Comment:
existing save() recovers from a missing file gracefully (line 99-104). There for you exception in line 108 will never be called. 

So first check if the file is there before line 99 and throw the exception there.